x-pack/plugins/apm/server/plugin.ts

```diff
@@ -141,13 +150,41 @@ export class APMPlugin implements Plugin<APMPluginSetup> {
       },
     });
 
+    const boundGetApmIndices = async () =>
+      getApmIndices({
+        savedObjectsClient: await getInternalSavedObjectsClient(core),
+        config: await mergedConfig$.pipe(take(1)).toPromise(),
+      });
+
     return {
       config$: mergedConfig$,
-      getApmIndices: async () =>
-        getApmIndices({
-          savedObjectsClient: await getInternalSavedObjectsClient(core),
-          config: await mergedConfig$.pipe(take(1)).toPromise(),
-        }),
+      getApmIndices: boundGetApmIndices,
+      createApmEventClient: async ({
+        request,
+        context,
+        debug,
+      }: {
+        debug?: boolean;
+        request: KibanaRequest;
+        context: RequestHandlerContext;
+      }) => {
+        const [indices, includeFrozen] = await Promise.all([
+          boundGetApmIndices(),
```

Comment:
Will the APM indices only be computed initially when `createApmEventClient` is called? Meaning, users should not re-use the return apm client across requests but create a new one each time?

Reply:
The APM indices will be refetched on every request, similar to our implementation. And yes, users should always create a new one each time. It needs a scoped cluster client and a Kibana request, so we are practically enforcing that.